<launch>
    <arg name="namespace_prefix" default="easy_handeye" />

    <arg name="robot_ip" doc="The IP address of the UR5 robot" />

    <arg name="marker_size" doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id" doc="The ID of the ArUco marker used" />


	<node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size" type="double" value="6" />
		<param name="max_new_marker_error" type="double" value="0.08" />
		<param name="max_track_error" type="double" value="0.2" />
		<param name="output_frame" type="string" value="camera_link" />

		<remap from="camera_image" to="/camera/color/image_raw" />
		<remap from="camera_info" to="/camera/color/camera_info" />
	</node>
		
    <!-- start the robot -->
    <!-- include file="$(find ur_bringup)/launch/ur5_bringup.launch">
        <arg name="limited" value="true" />
        <arg name="robot_ip" value="192.168.0.21" />
    </include -->
    <include file="$(find tm700_moveit_config)/launch/tm700_moveit_planning_execution.launch">
        <arg name="sim" value="false" />
        <arg name="robot_ip" value="192.168.0.200" />
    </include>

    <!-- start easy_handeye -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="false" />

        <arg name="tracking_base_frame" value="camera_link" />
        <arg name="tracking_marker_frame" value="ar_marker_2" />
        <arg name="robot_base_frame" value="base_link" />
        <arg name="robot_effector_frame" value="tip_link" />

        <arg name="freehand_robot_movement" value="false" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />

        <arg name="start_rviz" value="false" />
        <arg name="start_rqt" value="false" />

    </include>

</launch>
